services/src/main/java/org/keycloak/utils/SecureContextResolver.java

```diff
@@ -79,16 +78,15 @@ public static boolean isLocalAddress(String address) {
         if (address == null) {
             return false;
         }
-        // The host matches a CIDR notation of ::1/128
-        if (address.startsWith("[")) {
-            return LOCALHOST_IPV6.matcher(address).matches();
-        }
 
-        // The host matches a CIDR notation of 127.0.0.0/8
-        if (LOCALHOST_IPV4.matcher(address).matches()) {
-            return true;
+        if (NetUtil.isValidIpV4Address(address) || NetUtil.isValidIpV6Address(address)) {
+            try {
+                return InetAddress.getByName(address).isLoopbackAddress();
```

Comment:
I wonder if this might yield some false positives, does this match the full CIDR range of  `::1/128` and `127.0.0.0/8`, and nothing else? User agents are [rather strict](https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy) about what is allowed here, and might not set cookies properly if some other address is matched.

Reply:
> does this match the full CIDR range of `::1/128` and `127.0.0.0/8`, and nothing else?

Looking at the OpenJDK implementations for [v4](https://github.com/openjdk/jdk/blob/jdk-21%2B35/src/java.base/share/classes/java/net/Inet4Address.java#L189-L193) and [v6](https://github.com/openjdk/jdk/blob/jdk-21%2B35/src/java.base/share/classes/java/net/Inet6Address.java#L321-L327) addresses, it looks it matches exactly that. Question is if we can rely on it as the contract (the java doc) doesn't seem to say that and theoretically other JDK impls could have a different logic for that.

Reply:
Thanks for following up on that, I think we can consider the OpenJDK implementation sound enough in this regard. But yeah, the JavaDoc is rather limiting, not sure if there are other specifications we could reference.